partiql-spi/src/main/java/org/partiql/spi/function/RoutineProviderSignature.java

```diff
@@ -0,0 +1,59 @@
+package org.partiql.spi.function;
+
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * <p>
+ * This represents the signature of a routine provider. This is distinct from {@link RoutineSignature}, as it is specific
+ * to the provider, and not the instance. The provider signature is used to determine if a routine provider is applicable
+ * to a given call site, and if so, which routine provider to use.
+ * </p>
+ * <p>
+ * While this is currently the same as a {@link RoutineSignature}, additional methods <i>may</i> be added to this class
+ * for future use, such as for retrieving all representative signatures of an implementation (for debugging purposes
+ * or for error-messaging).
```

Comment:
from offline, these two classes, `RoutineProviderSignature` and `RoutineSignature`, differ. `RoutineProviderSignature` does not have a property for the return type, isNullCall, and isMissingCall. It only really uses the types of the input parameters.